packages/ngtools/webpack/src/transformers/import_factory.ts

```diff
@@ -0,0 +1,201 @@
+ * const routes: Routes = [{
+ *         path: 'lazy',
+ *         loadChildren: ɵ0
+ *     }];
+ * export class AppRoutingModule {
+ * }
+ * export { ɵ0 };
+ *
+ * The importFactory transformation will only see the AST after it is process by NGC.
+ * You can confirm this with the code below:
+ *
+ * const res = ts.createPrinter().printNode(ts.EmitHint.Unspecified, sourceFile, sourceFile);
+ * console.log(`### Original source: \n${sourceFile.text}\n###`);
+ * console.log(`### Current source: \n${currentText}\n###`);
+ *
+ * At this point it doesn't yet matter what the target (ES5/ES2015/etc) is, so the original
+ * constructs, like `class` and arrow functions, still remain.
+ *
+ */
+
+export function importFactory(
+  warningCb: (warning: string) => void,
+): ts.TransformerFactory<ts.SourceFile> {
+  return (context: ts.TransformationContext) => {
+    // Only transform files that contain `loadChildren`.
+    return (sourceFile: ts.SourceFile) => {
+      if (/\bloadChildren\b/.test(sourceFile.text)) {
+        const warning = `
+Found 'loadChildren' with a non-string syntax in ${sourceFile.fileName} but could not transform it.
+Make sure it matches the format below:
+
+loadChildren: () => import('IMPORT_STRING').then(m => m.EXPORT_NAME)
+
+Please note that only IMPORT_STRING and EXPORT_NAME can be replaced in this format.`;
+        const emitWarning = () => warningCb(warning);
+        const visitNode: ts.Visitor = (node: ts.Node) => {
+          // Only try to transform 'loadChildren' property assignments.
+          if (ts.isVariableDeclaration(node)) {
+            return replaceImport(node, context, emitWarning);
+          }
+
+          return ts.visitEachChild(node, visitNode, context);
+        };
+
+        return ts.visitNode(sourceFile, visitNode);
+      }
+
+      return sourceFile;
+    };
+  };
+}
+
+function replaceImport(
+  node: ts.VariableDeclaration,
+  context: ts.TransformationContext,
+  emitWarning: () => void,
+): ts.Node {
+  // This ONLY matches the original source code format below:
+  // loadChildren: () => import('IMPORT_STRING').then(m => m.EXPORT_NAME)
+  // And expects that source code to be transformed by NGC (see comment for importFactory).
+  // It will not match nor alter variations, for instance:
+  // - not using arrow functions
+  // - not using `m` as the module argument
+  // - using `await` instead of `then`
+  // - using a default export (https://github.com/angular/angular/issues/11402)
+  // The only parts that can change are the ones in caps: IMPORT_STRING and EXPORT_NAME.
+
+  // Exit early if the structure is not what we expect.
+
+  // ɵ0 = something
+  const name = node.name;
+  if (!(
+    ts.isIdentifier(name)
+    && /ɵ\d+/.test(name.text)
+  )) {
+    return node;
+  }
+
+  const initializer = node.initializer;
+  if (initializer === undefined) {
+    return node;
+  }
+
+  // ɵ0 = () => something
+  if (!(
+    ts.isArrowFunction(initializer)
```

Comment:
What about if target is es5?

Reply:
The target does not make a difference. This transform operates over the result from NGC and before any downleveling done by TS proper.